### ISSUE TITLE
Squash migrations for better performance

### ISSUE THREAD

**Issue by matmair** [inventree/InvenTree#5015]
We currently ship a lot of migrations that are run in order; even on new instances. I suggest shipping 1.0 with [squashed migrations](https://docs.djangoproject.com/en/dev/topics/migrations/#migration-squashing) and removing the old migrations with the next version. There would need to be a check that versions below 0.x can not be upgraded to a version > 1.x

**Comment by matmair** [comment]
I made a [POC](https://github.com/matmair/InvenTree/tree/squash-migrations) of how that could look like. For this to happen we need at least a week or so of database change freeze to
a) make the patch (will take a few hours probably)
b) have the change running for a few days to be sure it does not destroy instance

**Comment by SchrodingersGat** [comment]
@matmair that looks nice, I was expecting we'd have to have a breaking change here but, if I read it correctly, this is a "drop in" replacement for an existing setup?

I'd like to ensure we have https://github.com/inventree/InvenTree/pull/4973 working before this point

**Comment by matmair** [comment]
@SchrodingersGat yes, that is the idea. Will speed up startups as well in my experience.

**Comment by matmair** [comment]
@SchrodingersGat I would start working on this if there are no objections. Is this clear to go?

**Comment by SchrodingersGat** [comment]
@matmair sounds good :) We will have to treat this one very carefully

**Comment by matmair** [comment]
Full transparency: I did not get this working in 4 or 5 attempts due to circular import issue and wired postgres warnings.
Maybe we need to simplify a few dependencies between apps and start out fresh with something like https://django-remake-migrations.readthedocs.io

**Comment by SchrodingersGat** [comment]
Yes it's a tough one for sure. Let's reconsider, it's still "workable" even if there are performance improvements to be had.

**Comment by SchrodingersGat** [comment]
New attempt in https://github.com/inventree/InvenTree/pull/12830
